src/iologindata.cpp

```diff
@@ -44,6 +44,32 @@ Account IOLoginData::loadAccount(uint32_t accno)
 	return account;
 }
 
+bool IOLoginData::addAccountStorageValue(const uint32_t accno, const uint32_t key, const int32_t value)
+{
+	std::string query;
+
+	if (value != -1) {
+		query = fmt::format("INSERT INTO `account_storages` (`account_id`, `key`, `value`) VALUES ({:d}, {:d}, {:d}) ON DUPLICATE KEY UPDATE `value` = {:d}", accno, key, value, value);
+	} else {
+		query = fmt::format("DELETE FROM `account_storages` WHERE `account_id` = {:d} AND `key` = {:d}", accno, key);
+	}
+
+	return Database::getInstance().executeQuery(query);
+}
+
+bool IOLoginData::getAccountStorageValue(const uint32_t accno, const uint32_t key, int32_t& value)
```

Comment:
I don't think that returning the value by referenced parameter is good. Return value does not have to be a boolean at all, you can distinguish missing storage (-1) from set storage (>= 0) easily.

Anyway, this is probably true for players storage too, I don't see how mixing int32 with uint32 for value is acceptable.

Reply:
I agree with you indeed. Do you think it is a good idea I change the `player:setStorageValue` implementation too within this PR?

Reply:
I'd consider changing return type a breaking change and I have no idea what the policy here is (if any). This should probably be done in [2.0](https://github.com/otland/forgottenserver/projects/3).